recipes/repository.rb

```diff
@@ -47,8 +47,8 @@
     action :remove
   end
 when 'rhel', 'fedora', 'amazon'
-  # Import new RPM key
-  if node['datadog']['yumrepo_gpgkey_new']
+  # Import new RPM key if installing Agent 6
+  if node['datadog']['agent6']
```

Comment:
I think we should keep the test on `node['datadog']['yumrepo_gpgkey_new']` just because we also use the value in `node['datadog']['yumrepo_gpgkey_new']` to download the key, thus, we don't want it to be null or empty.